injector/grpc_disruption.go

```diff
@@ -98,10 +102,11 @@ func (i *GRPCDisruptionInjector) Clean() error {
 	return conn.Close()
 }
 
-func connectToServer(serverAddr string) (*grpc.ClientConn, error) {
+func connectToServer(serverAddr string, timeout time.Duration) (*grpc.ClientConn, error) {
 	var opts []grpc.DialOption
 	opts = append(opts, grpc.WithInsecure()) // Future Work: make secure
 	opts = append(opts, grpc.WithBlock())
+	opts = append(opts, grpc.WithTimeout(timeout))
 
 	conn, err := grpc.Dial(serverAddr, opts...)
```

Comment:
You might want to use `DialContext` here and create a context with deadline to manage the timeout.